lib/src/rules/parameter_assignments.dart

```diff
@@ -88,9 +132,25 @@ class _Visitor extends SimpleAstVisitor {
       // Getters don't have parameters.
       parameterList.parameters.forEach((e) {
         if (node.body.isPotentiallyMutatedInScope(e.element)) {
-          rule.reportLint(e);
+          _reportIfSimpleParameterOrWithDefaultValue(e, node);
         }
       });
     }
   }
+
+  void _reportIfSimpleParameterOrWithDefaultValue(
+      FormalParameter parameter, AstNode functionOrMethodDeclaration) {
+    final shouldReport = parameter is SimpleFormalParameter ||
+        DartTypeUtilities
+            .traverseNodesInDFS(functionOrMethodDeclaration)
+            .where((n) =>
+                _isDefaultFormalParameterWithDefaultValue(parameter) ||
```

Comment:
Can we move at least part of this test outside the enclosing closure? If the parameter isn't a default formal parameter without a default value, then there's no reason to traverse everything in the function body.